File: Dart/src/krishna/miniproject/FacultyUser.dart
import 'User.dart';

class FacultyUser extends User{

  String _subject;

  FacultyUser(String name, String userId, String subject)
      : _subject=subject,
        super(name, userId);

  @override
  void showDetails(){
    print("Name: $name");
    print("User Id: $userId");
    print("Subject: $_subject");
  }

}

File: src/com/krishna/miniproject/FacultyUser.java
package com.krishna.miniproject;

public class FacultyUser extends User {
    private String subject;

    public FacultyUser(String name, String userId, String subject) {
        super(name, userId);
        this.subject = subject;
    }
    @Override
    public void showDetails(){
        System.out.println("Name: "+ name);
        System.out.println("User Id: "+ userId );
        System.out.println("Subject: "+subject);
    }

}
